src/plugins/vis_type_timelion/server/usage_collector/get_usage_collector.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { getIndexArgs } from '../../common/parser';
+import { getIndexPatternsService } from '../services';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string }; updated_at: string }>;
+
+export interface TimelionUsage {
+  timelion_use_scripted_fields_total: number;
+}
+
+const getPastDays = (dateString: string): number => {
+  const date = new Date(dateString);
+  const today = new Date();
+  const diff = Math.abs(date.getTime() - today.getTime());
+  return Math.trunc(diff / (1000 * 60 * 60 * 24));
+};
+
+export const getStats = async (
+  esClient: ElasticsearchClient,
+  soClient: any,
+  index: string
+): Promise<TimelionUsage | undefined> => {
+  const indexPatternsServiceFactory = getIndexPatternsService();
+  const indexPatternsService = await indexPatternsServiceFactory(soClient, esClient);
```

Comment:
honestly I'm not sure about using index patterns service for telemetry. Maybe we can get scripted fields directly from so. 

@stratoula what do you think?

Reply:
Yes I prefer that way too 👍

Reply:
Okay, I changed it and now use soClient.